src/state-changes/state.changes.consumer.service.ts

```diff
@@ -0,0 +1,284 @@
+import { CompetingRabbitConsumer } from "src/common/rabbitmq/rabbitmq.consumers";
+import { BlockWithStateChangesRaw, ESDTType, StateChanges } from "./entities";
+import { AccountDetails, AccountDetailsRepository } from "src/common/indexer/db";
+import { Inject, Injectable } from "@nestjs/common";
+import { CacheService } from "@multiversx/sdk-nestjs-cache";
+import { CacheInfo } from "src/utils/cache.info";
+import { TokenType } from "src/common/indexer/entities";
+import { NftType } from "src/endpoints/nfts/entities/nft.type";
+import { NftSubType } from "src/endpoints/nfts/entities/nft.sub.type";
+import { ClientProxy } from "@nestjs/microservices";
+import { StateChangesDecoder } from "./utils/state-changes.decoder";
+import { AddressUtils, OriginLogger } from "@multiversx/sdk-nestjs-common";
+
+@Injectable()
+export class StateChangesConsumerService {
+    private readonly logger = new OriginLogger(StateChangesConsumerService.name);
+    static SYSTEM_ACCOUNTS = [
+        "erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqllls0lczs7", // stakingScAddress
+        "erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqplllst77y4l", // validatorScAddress
+        "erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqzllls8a5w6u", // esdtScAddress 
+        "erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqrlllsrujgla", // governanceScAddress
+        "erd1qqqqqqqqqqqqqqqpqqqqqqqqqrlllllllllllllllllllllllllsn60f0k", // jailingAddress 
+        "erd1qqqqqqqqqqqqqqqpqqqqqqqqlllllllllllllllllllllllllllsr9gav8", // endOfEpochAddress
+        "erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqylllslmq6y6", // delegationManagerScAddress
+        "erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqq0llllsqkarq6", // firstDelegationScAddress
+        "erd1qqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqq6gq4hu", // contractDeployScAdress
+        "erd17rc0pu8s7rc0pu8s7rc0pu8s7rc0pu8s7rc0pu8s7rc0pu8s7rcqqkhty3", // genesisMintingAddress
+        "erd1lllllllllllllllllllllllllllllllllllllllllllllllllllsckry7t", // systemAccountAddress
+        "erd1llllllllllllllllllllllllllllllllllllllllllllllllluqq2m3f0f", // esdtGlobalSettingsAddresses[0] 
+        "erd1llllllllllllllllllllllllllllllllllllllllllllllllluqsl6e366", // esdtGlobalSettingsAddresses[1]
+        "erd1lllllllllllllllllllllllllllllllllllllllllllllllllupq9x7ny0", // esdtGlobalSettingsAddresses[2] 
+    ];
+
+    constructor(
+        private readonly cacheService: CacheService,
+        private readonly accountDetailsRepository: AccountDetailsRepository,
+        @Inject('PUBSUB_SERVICE') private clientProxy: ClientProxy,
+    ) { }
+
+    @CompetingRabbitConsumer({
+        exchange: 'state_accesses',
+        queueName: 'api_state_accesses_queue-test',
```

Comment:
is there a way to provide these from config? not sure if the decorator supports variables